### DISCUSSION TITLE
Security advice

### DISCUSSION THREAD

**Discussion by joedixon** [livewire/livewire#4090] [1. Help]
Hi 👋 

I work on an application which uses Livewire. Recently, the application seeemed to be the subject of a malicious user who, on the face of it, seemed to be targeting our Livewire components.

The project error tracker was filled up overnight with lots of Livewire based exceptions such as:

- `Livewire encountered corrupt data when trying to hydrate...`
- `Unable to call component method. Public method [random method name] not found on component`
- `Unable to find component: [random component name]`

I have read the security section of the Livewire docs and, as far as I can tell, Livewire has done it's job and recognising the malicious intent and guarding the application against it, which is great.

I'm just wondering if anyone else has experienced anything similar and/or whether there is anything else we can do to further protect ourselves?

**Comment by joshhanley** [comment]
@joedixon yeah looks like Livewire is doing what it was meant to!

I haven't experienced an attack, but there are a few things you can do.

Make sure to change all methods to `protected` except for those that should be called from `wire:click` which need to be public.

For those few methods that are public, make sure you have authorisation in them.

Same for properties, only make public the properties that should be available to the front end (for the user to edit or to maintain state), the rest make protected.

Also you can double check to make sure if they submit data manually via javascript using `$wire.set('property', 'some random value')` (instead of using form fields) that all data is validated and sanitised before it is used or persisted.

That's all I can think of off the top of my head.

The Livewire end point is secured by the checksum, which requires your app key to generate a valid checksum, so that should be safe (hence the "corrupt data" error probably).

Hope this helps!
